### DISCUSSION TITLE
How do I add an extension plugin?

### DISCUSSION THREAD

**Discussion by dominikwilkowski** [zed-industries/zed#12621] [Feature Requests]
I'd love to add a plugin that shows things inside the editor per line.

Looking at [the extensions repo](https://github.com/zed-industries/extensions/blob/main/AUTHORING_EXTENSIONS.md) it mentions so far the only things supported are syntax, language and themes.

Looking at [the roadmap](https://zed.dev/roadmap) under Adoption it mentions "Plugins" under "Extensibility" which I assume might be the term we use for this?

Is that a thing yet? If so are there any docs yet? Sorry if I missed it.

**Comment by SomeoneToIgnore** [comment]
https://github.com/zed-industries/zed/blob/1a0fd730325459ed6b81c459498d90883e2a35eb/crates/extension_api/src/extension_api.rs#L59-L118

Currently, this is all any extension is able to do programmatically, so there's no editor-related API yet and extensions cannot directly influence what is displayed on the screen.

So seems like it's impossible now.
